client/client.go

```diff
@@ -65,6 +66,13 @@ type Attachment struct {
 	Owner   string `json:"-"` // IP address of uploader, used for rate limiting
 }
 
+// Icon represents a message icon
+type Icon struct {
+	URL  string `json:"url"`
+	Type string `json:"type,omitempty"`
+	Size int64  `json:"size,omitempty"`
```

Comment:
I don't see anywhere in the code where Size or Type are filled/populated. Or am I missing something? I would have made icon just a string.